src/Raven.Client/Documents/DocumentStoreBase.cs

```diff
@@ -73,6 +73,7 @@ public virtual void ExecuteIndex(AbstractIndexCreationTask task)
         /// </summary>
         public virtual Task ExecuteIndexAsync(AbstractIndexCreationTask task, CancellationToken token = default(CancellationToken))
         {
+            AssertInitialized();
```

Comment:
Are there any additional locations where this can cause problems for us?

Reply:
We might. If the operation requires a database it can cause a problem.
Will start the review.
http://issues.hibernatingrhinos.com/issue/RavenDB-8700